azurerm/internal/services/web/parse/certificate_binding.go

```diff
@@ -0,0 +1,56 @@
+package parse
+
+import (
+	"fmt"
+	"strings"
+)
+
+// This is a special case ID for a meta resource that links a web certificate to a Web App or Function App
+
+type CertificateBindingId struct {
+	HostnameBindingId
+	CertificateId
+}
+
+func NewCertificateBindingId(hostnameBinding string, certificate string) (CertificateBindingId, error) {
```

Comment:
as a general rule, if this returns a secondary value of `error` the first item should be a pointer, which means that if an error is returned and you try and access the first object, it'll be nil and so the program will crash; meaning these'll get caught at test time:

```suggestion
func NewCertificateBindingId(hostnameBinding string, certificate string) (*CertificateBindingId, error) {
```